### PR TITLE
Feature/compare testrail and local tests structure

### PR DESCRIPTION
**Pull request by vladimir-s-quantory**
Fix: fixed bug that prevented folder structure creation on Linux
Feat: changed logic of SectionSynchronizer so it calls API only once when app is started. New sections are added to structure without API calls

### CODE REVIEW

**Comment by Brrovko** [comment]
Please, update AssemblyVersion in GherkinSyncTool.csproj

**Comment by Brrovko** [comment]
Please, update AssemblyVersion in GherkinSyncTool.csproj

**Review by Georgiy-Petrov — APPROVED** [review]

File: GherkinSyncTool/GherkinSyncTool/Synchronizers/TestRailSynchronizer/Model/CreateSectionRequest.cs [hunk 1]
```diff
@@ -6,6 +6,16 @@ public class CreateSectionRequest
         public string Description { get; init; }
         public ulong? ParentId { get; init; }
         public ulong SuiteId { get; init; }
-        public string Name { get; init; }  
+        public string Name { get; init; }
+
+        public CreateSectionRequest() { }
```
Comment (Brrovko) [inline GherkinSyncTool/GherkinSyncTool/Synchronizers/TestRailSynchronizer/Model/CreateSectionRequest.cs:11]: Not used.

File: GherkinSyncTool/GherkinSyncTool/Synchronizers/TestRailSynchronizer/Content/SectionSynchronizer.cs [hunk 2]
```diff
@@ -61,11 +70,10 @@ public ulong GetOrCreateSectionId(string path)
         {
```
Comment (Brrovko) [inline GherkinSyncTool/GherkinSyncTool/Synchronizers/TestRailSynchronizer/Content/SectionSynchronizer.cs:70]: Let's specify what path expected?
`/// <param name="relativePath">Relative path to .feature file</param>
`
`public ulong GetOrCreateSectionId(string relativePath) `

File: GherkinSyncTool/GherkinSyncTool/Synchronizers/TestRailSynchronizer/Content/SectionSynchronizer.cs [hunk 3]
```diff
@@ -78,14 +86,18 @@ public ulong GetOrCreateSectionId(string path)
         /// <param name="projectId">TestRail project Id</param>
         /// <param name="sectionId">TestRail section Id, null for the tests root</param>
         /// <returns>Section Id for the selected .feature file</returns>
-        private ulong GetOrCreateSectionIdRecursively(IEnumerable<TestRailSection> targetSections, Queue<string> sourceSections, 
+        private ulong GetOrCreateSectionIdRecursively(List<TestRailSection> targetSections,
+            Queue<string> sourceSections,
             ulong suiteId, ulong projectId, ulong? sectionId = null)
```
Comment (Brrovko) [inline GherkinSyncTool/GherkinSyncTool/Synchronizers/TestRailSynchronizer/Content/SectionSynchronizer.cs:91]: No method use `ulong? sectionId argument`

File: GherkinSyncTool/GherkinSyncTool/Synchronizers/TestRailSynchronizer/Content/SectionSynchronizer.cs [hunk 4]
```diff
@@ -94,15 +106,35 @@ public ulong GetOrCreateSectionId(string path)
...
+        {
+            var newSection =
+                new TestRailSection
+                {
+                    Id = sectionId,
+                    SuiteId = suiteId,
+                    ParentId = parentId,
+                    Name = folderName
+                };
+            targetSections.Add(newSection);
```
Comment (Brrovko) [inline GherkinSyncTool/GherkinSyncTool/Synchronizers/TestRailSynchronizer/Content/SectionSynchronizer.cs:136]: Didn't catch the logic here.

**Review by Brrovko — CHANGES_REQUESTED** [review]

**Review by Georgiy-Petrov — APPROVED** [review]

File: GherkinSyncTool/GherkinSyncTool/FeatureParser/FeatureParser.cs [hunk 5]
```diff
@@ -21,7 +21,7 @@ public List<IFeatureFile> Parse(IEnumerable<string> gherkinFilePaths)
                 FeatureFile file = new FeatureFile
                 {
                     Document = ParseFeatureFile(gherkinFilePath),
-                    RelativePath = gherkinFilePath
+                    AbsolutePath = gherkinFilePath
```
[the remark this answers is not in the document]
Comment (vladimir-s-quantory) [reply GherkinSyncTool/GherkinSyncTool/FeatureParser/FeatureParser.cs:24]: Please, check your apspettings.json file for the value of the BaseDirectory parameter. Is it full or relative to the app location?

File: GherkinSyncTool/GherkinSyncTool/Configuration/Config.cs [hunk 6]
```diff
@@ -12,13 +15,17 @@ public class Config
         public string TestRailBaseUrl { get; set; } 
         public string TestRailUserName { get; set; }
         public string TestRailPassword { get; set; }
+        public string TestCasesDirectoryName { get; set; }
```
Comment (Brrovko) [inline GherkinSyncTool/GherkinSyncTool/Configuration/Config.cs:18]: Not used

File: GherkinSyncTool/GherkinSyncTool/Configuration/Config.cs [hunk 7]
```diff
@@ -12,13 +15,17 @@ public class Config
...
         private string _directory;
         public string BaseDirectory
         {
-            get => _directory ?? Directory.GetCurrentDirectory();
+            get => _directory;
             set
             {
-                _directory = Path.GetRelativePath(Directory.GetCurrentDirectory(), value);
+                var info = new DirectoryInfo(value);
```
Comment (Brrovko) [inline GherkinSyncTool/GherkinSyncTool/Configuration/Config.cs:25]: Check please value for null

**Review by Brrovko — APPROVED** [review]

**Review by Brrovko — APPROVED** [review]

**Review by Georgiy-Petrov — APPROVED** [review]

**Review by Georgiy-Petrov — APPROVED** [review]

**Review by SergeyRuzin — APPROVED** [review]

**Pull request merged**